### PR TITLE
Standalone action

### PR DESCRIPTION
**Pull request by assignUser**
I have created the standalone action and updated the files in `inst` accordingly.
I tested this here: https://github.com/assignUser/simstudy/runs/4740711766?check_suite_focus=true
Maybe  @lorenzwalthert you can test it with {styler}? But please update your RSPM link, see: https://community.rstudio.com/t/pak-lockfile-creation-within-github-action-hangs-forever/125813/7

### CODE REVIEW

**Comment by assignUser** [comment]
oh we probably should update the documentation too

File: inst/touchstone-comment.yaml [hunk 1]
```diff
@@ -15,38 +15,4 @@ jobs:
...
-            var fs = require('fs');
-            var issue_number = Number(fs.readFileSync('./NR'));
-            var body = fs.readFileSync('./info.txt').toString();
-            await github.issues.createComment({
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              issue_number: issue_number,
-              body: body
-            });
+      - uses: lorenzwalthert/touchstone/actions/comment@standalone-action
```
Comment (lorenzwalthert) [inline inst/touchstone-comment.yaml:18]: That way, I am not sure we can't version the action can we? Wouldn't it make more sense like this?
```
lorenzwalthert/touchstone/actions/comment@main
```

**Comment by lorenzwalthert** [comment]
Great :D. I love seeing the workflow file shrink 😄

File: inst/touchstone-comment.yaml (hunk 1 above)
Comment (assignUser) [reply inst/touchstone-comment.yaml:18]: ah well this was mainly for testing (as the actions are not on main yet), we could then create a tag so people can use a fixed version of the action

Comment (lorenzwalthert) [reply inst/touchstone-comment.yaml:18]: ah ok I see. Won't have time now, but will test it tomorrow.

Comment (lorenzwalthert) [reply inst/touchstone-comment.yaml:18]: (I first did not get that the part after `@` is the branch)

**Comment by lorenzwalthert** [comment]
I
* removed the cancel action, as GitHub now supports in-flight cancelling and {styler}'s touchstone script [already used](https://github.com/r-lib/styler/pull/898/files#diff-6a214d1d476575d0081301c758ecf219b9d1eb92024e7002f4edd9e440c85e83L3) it. Maybe we can also add the same cancelling to the template?
* updated rspm and other config in `touchstone/config.json`

**Comment by lorenzwalthert** [comment]
Thanks for fixing the tests, the concurrency thing and the added README.md 👍. Maybe it makes more sense to link this article to the security concerns? https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

Also, our bash scripts are not save against injection as they don't use quotes for env variables, e.g. [here](https://github.com/lorenzwalthert/touchstone/pull/92/files#diff-cddbfc360a6fcd405c3a68abac199e155878e57ac192c119337659cddfe0a27fR32).

Also, I wonder if we still need two workflows, since in the [article you linked](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-action-instead-of-an-inline-script-recommended), they say one strategy to avoid injections is to use an action instead of a script.

File: actions/receive/action.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      with:
+         fetch-depth: 0
+    - name: Set up git user
+      run: |
+        git config --local user.name "GitHub Actions"
+        git config --local user.email "actions@github.com"
+      shell: bash
+    - name: Ensure base branch is fetched
+      run: |
+        git checkout -b $GITHUB_HEAD_REF # this is current ref. This is required for naming.
```
Comment (lorenzwalthert) [inline actions/receive/action.yaml:32]:
```suggestion
        git checkout -b "$GITHUB_HEAD_REF" # this is current ref. This is required for naming.
```

**Comment by assignUser** [comment]
> Also, I wonder if we still need two workflows, since in the [article you linked](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-an-action-instead-of-an-inline-script-recommended), they say one strategy to avoid injections is to use an action instead of a script.

I think those are two separate issues code injection via context and r&w access to our repo while also executing code supplied by a (malicious) third party. So I would say we stick with 2. I will edit the readme in a moment, should I adjust the templates to use `@main` or do you want to change that after merging?

**Comment by lorenzwalthert** [comment]
Sounds good. You can change it before the merge. Thanks.

**Comment by assignUser** [comment]
:heavy_check_mark: :D

**Review by lorenzwalthert — APPROVED** [review]
You can merge it then.

**Pull request merged**